lib/gutentag/tagged_with_query.rb

```diff
@@ -35,11 +31,13 @@ def name_query
   end
 
   def query
-    ids? ? id_query : name_query
-  end
-
-  def tag_ids
-    options[:ids] || Array(options[:tags]).collect(&:id)
+    if ids = options[:ids].presence
```

Comment:
There are two things making me pause about this PR, and they're both visible in this line.

I don't like assigning to variables within the if clause - unnecessary too, given they're only being used in the following lines, so better to just consistently refer to the particular option directly from `options`.

The second issue is that the three branches of logic here (if/elsif/else) feels a little complex… though I'm far less sure what a better approach is. I'm going to mull this over a little more.